### PR TITLE
changelog for edge-21.7.4

### PR DESCRIPTION
**Pull request by dadjeibaah**
This release continues to focus on dependency updates. It also adds proxy a
change that add a new `l5d-proxy-error` information header to distinguish
proxy generated errors from application generated errors. It also adds
support for proxies in gateway mode to now have `endpoints` as gateway targets.

* Updated several project dependencies
* Added support for gateway proxies to use `endpoint` targets in addition to
  using logical services
* Added a new `l5d-proxy-error` on responses that allows proxy-generated error
  responses to be distinguished from application-generated error responses.
* Removed support for configuring `HTTP/2` keepalives via the proxy.
  Configuring this setting would sometimes cause conflicts with Go gRPC servers
  and clients
* Added a new `target_addr` label to `*_tcp_accept_errors` metrics to improve
  diagnostics, especially for TLS detection timeouts

Signed-off-by: Dennis Adjei-Baah <dennis@buoyant.io>

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
 
+## edge-21.7.4
+
+This release continues to focus on dependency updates. It also adds proxy a
+change that add a new `l5d-proxy-error` information header to distinguish
+proxy generated errors from application generated errors. It also adds
+support for proxies in gateway mode to now have `endpoints` as gateway targets.
+
+* Updated several project dependencies
+* Added support for gateway proxies to use `endpoint` targets in addition to
```
Comment (olix0r) [inline CHANGES.md:11]: let's omit this for now -- it's only user-facing when Matei's PR lands to support statefulsets
```suggestion
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+  using logical services
+* Added a new `l5d-proxy-error` on responses that allows proxy-generated error
+  responses to be distinguished from application-generated error responses.
+* Removed support for configuring `HTTP/2` keepalives via the proxy.
```
Comment (olix0r) [inline CHANGES.md:15]:
```suggestion
* Removed support for configuring HTTP/2 keepalives via the proxy.
```

**Review by olix0r — APPROVED** [review]

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,20 @@
 # Changes
 
+## edge-21.7.4
+
+This release continues to focus on dependency updates. It also adds proxy a
```
Comment (kleimkuhler) [inline CHANGES.md:5]: nit: Fine to keep this, but I think from now one we're always going to have quite a few dependency updates so it's probably not worth mentioning in every edge.

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+change that add a new `l5d-proxy-error` information header to distinguish
```
Comment (kleimkuhler) [inline CHANGES.md:6]: nit: feels a little redundant to say it adds a change that added a further change

```suggestion
This release continues to focus on dependency updates. It also adds the
`l5d-proxy-error` information header to distinguish proxy generated errors
```

**Review by kleimkuhler — APPROVED** [review]
nits but otherwise 👍

**Pull request merged**